### PR TITLE
Fixing build traversal projects and setting up a pipeline to publish a new release to github

### PR DESCRIPTION
**Pull request by sorosoro2**
Added Build.Traversal.targets, and Build.Props, to ensure that the compilation works properly on dirs.proj

Added a pipeline with 6 tasks
1. Install Nuget 4.3.0 to clear dependency issues between .NET standard 2.0 used in azure function and .NET framework 4.7.2
2. Nuget restore
3. msbuild
4. Run unit tests
5. Archive bin
6. Create github release

### CODE REVIEW

File: publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        clean: true
+
+    - task: ArchiveFiles@2
+      displayName: 'Archive azure function'
+      inputs:
+        rootFolderOrFile: AzureFunction/bin/Release/netstandard2.0/
+        includeRootFolder: false
+        archiveFile: '$(Build.ArtifactStagingDirectory)/$(Build.BuildNumber).zip'
+
+    - task: GitHubRelease@1
```
Comment (vithati) [inline publish.yml]: Add flag to publish or not

File: publish.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    demands: msbuild
+
+  steps:
+    - task: MSBuild@1
+      displayName: 'Build dirs.proj'
+      inputs:
+        solution: dirs.proj
+        configuration: Release
+        clean: true
+
```
Comment (vithati) [inline publish.yml]: Add L0 test cases task.

**Review by vithati — CHANGES_REQUESTED** [review]

File: publish.yml (hunk 1 above)
Comment (sorosoro2) [reply publish.yml]: Added a variable check for this

File: publish.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        testAssemblyVer2: 'Tests/bin/Release/**/*.dll'
+    
+    - task: ArchiveFiles@2
+      displayName: 'Archive azure function'
+      inputs:
+        rootFolderOrFile: AzureFunction/bin/Release/netstandard2.0/
+        includeRootFolder: false
+        archiveFile: '$(Build.ArtifactStagingDirectory)/$(Build.BuildNumber).zip'
+
+    - task: GitHubRelease@1
```
Comment (vithati) [inline publish.yml]: Can you check later, how can we mark private, beta, alpha and release version.

File: publish.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      inputs:
+        testAssemblyVer2: 'Tests/bin/Release/**/*.dll'
+    
+    - task: ArchiveFiles@2
+      displayName: 'Archive azure function'
+      inputs:
+        rootFolderOrFile: AzureFunction/bin/Release/netstandard2.0/
+        includeRootFolder: false
+        archiveFile: '$(Build.ArtifactStagingDirectory)/$(Build.BuildNumber).zip'
+
```
Comment (vithati) [inline publish.yml]: Can you add one more step to upload zip has build artifact. Mostly for testing we can download from there and validate.

**Review by vithati — CHANGES_REQUESTED** [review]

**Review by vithati — APPROVED** [review]

**Pull request merged**